packages/react-badge/src/components/CounterBadge/CounterBadge.types.ts

```diff
@@ -1,71 +1,49 @@
 import type { BadgeProps, BadgeState } from '../Badge/index';
 
-/**
- * {@docCategory CounterBadge}
- */
-export interface CounterBadgeProps extends Omit<BadgeProps, 'appearance' | 'shape'> {
-  /**
-   * A Badge can be circular or rounded
-   * @defaultvalue circular
-   */
-  shape?: Extract<BadgeProps['shape'], 'rounded' | 'circular'>;
-
-  /**
-   * A Badge can be filled, ghost
-   * @defaultvalue filled
-   */
-  appearance?: Extract<BadgeProps['appearance'], 'filled' | 'ghost'>;
-
+export interface CounterBadgeCommons {
   /**
    * Max number to be displayed
-   * @defaultvalue 99
+   * @default 99
    */
-  overflowCount?: number;
+  overflowCount: number;
 
   /**
    * Value displayed by the Badge
-   * @defaultvalue 0
+   * @default 0
    */
-  count?: number;
+  count: number;
 
   /**
    * If the badge should be shown when count is 0
-   * @defaultvalue false
+   * @default false
    */
-  showZero?: boolean;
+  showZero: boolean;
 
   /**
    * If a dot badge should be displayed
-   * @defaultvalue false
+   * @default false
    */
-  dot?: boolean;
-}
+  dot: boolean;
 
-/**
- * {@docCategory CounterBadge}
- */
-export interface CounterBadgeState extends BadgeState {
   /**
-   * Max number to be displayed
-   * @defaultvalue 99
+   * A Badge can be circular or rounded
+   * @default circular
    */
-  overflowCount: number;
+  shape: Extract<BadgeProps['shape'], 'rounded' | 'circular'>;
```

Comment:
My thought was that it would be more consistent and easier to maintain if extracted from the original type

Reply:
FWIW I don't think this is better for maintenance :). Extract doesn't do any type checking, nor has any compile errors if the types change. The resulting type of CounterBadge's `shape` prop would change if Badge's `shape` prop changed, but it wouldn't cause a compile error in the type definition, and then you'd be left with a potentially confusing Extract definition.

I.e. this compiles `Extract<('foo' | 'bar'), ('foo' | 'qux')>`, and the resulting type is `'foo'`

However, with my suggestion, you _would_ get a compile error if the types were changed and made incompatible in the future, because the call to `useBadge` from `useCounterBadge` would fail on the incompatible types.

Reply:
yeah wasn't aware that `Extract` does not type check, I've removed it